platform/kube/inject/initializer.go

```diff
@@ -162,29 +162,38 @@ func NewInitializer(restConfig *rest.Config, config *Config, cl kubernetes.Inter
 	return i, nil
 }
 
-func (i *Initializer) initialize(in interface{}, patcher patcherFunc) error {
-	obj, err := meta.Accessor(in)
-	if err != nil {
-		return err
+func modifyRequired(objNamespace string, ignore, exclude, include []string) bool {
```

Comment:
The purpose of this function is a bit confusing, because there is also the injectRequired, which already has control logic if to inject or not based on the namespaces include exclude etc.

I thought the initializer would need to remove itself from the list of initializers no matter what. So the logic would be like:
if injectRequired(c.Policy, obj) {
   inject()
} else {
    glog.V(2).Infof("Skipping %s/%s due to policy check", obj.GetNamespace(), obj.GetName())
}
removeFromPending()

Reply:
Do we want to retain the option of running multiple initializers in the same cluster? If not, we can remove the "off" policy and make the initializer only operate cluster-wide. That would simplify some of this code.

If concurrent initializers is supported, then initializer needs to do one of three things: 

1) ignore a deployment completely (and let another initializef handle it)
2) remove itself from the list of pending initialiers (and do nothing; ignored/excluded namespaces)
3) (2) plus injection the sidecar 

`modifyRequired` checks if (2) is required.
`injectRequired` checks if (3) is required.

Reply:
Got it, thanks for clarifying.